src/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,215 @@
+import React, {useState, useEffect} from "react";
+//@ts-ignore
+import {default as ReactDatePicker, registerLocale} from "react-datepicker";
+import enGB from "date-fns/locale/en-GB";
+import StyledDatePickerWrapper from "./DatePicker.styles";
+import Popover from "../Popover";
+import Button from "../Button";
+import Box from "../Box";
+import ClickAwayListener from "../ClickAwayListener";
+import Typography from "../Typography";
+import {customStyles} from "../../core/styleFunctions";
+import {Modifier} from "react-popper";
+import {Placement} from "@popperjs/core";
+
+export interface DatePickerProps {
+  defaultDate?: Date | null;
+  defaultTime?: string;
+  /**
+   * Callback fired when the date changes
+   */
+  onChange?: (date: Date | null) => void;
+  onTimeChange?: (time: string, date: Date | null) => void;
+  onClose?: () => void;
+  popoverCs?: customStyles;
+  isBlock?: boolean;
+  hasTime?: boolean;
+  modifiers?: ReadonlyArray<Modifier<any>>;
+  placement?: Placement;
+  anchorEl?: React.RefObject<HTMLElement>;
+  "data-cy"?: string;
+  "data-cy-today"?: string;
+  "data-cy-clear"?: string;
+}
+
+interface TimePickerProps {
+  selectedTime: string; // Assuming selectedTime is a string in the format "HH:mm"
+  selectedDate?: Date | null;
+  onTimeChange: (time: string, date: Date | null) => void;
+}
+
+const TimePicker: React.FC<TimePickerProps> = ({
+  selectedTime,
+  selectedDate = null,
+  onTimeChange,
+}) => {
+  const [inputValue, setInputValue] = useState(selectedTime);
+
+  useEffect(() => {
+    if (selectedTime && selectedTime.includes("NA")) {
+      setInputValue("");
+    } else {
+      setInputValue(selectedTime.substring(0, 5));
+    }
+  }, [selectedTime]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTime = event.target.value;
+    setInputValue(newTime);
+
+    if (onTimeChange) {
+      onTimeChange(newTime, selectedDate);
+    }
+  };
+
+  return (
+    <Box
+      cs={{
+        display: "flex",
+        marginBottom: "spacing.sm",
+        marginLeft: "spacing.sm",
+        alignItems: "center",
+      }}
+    >
+      <Typography variant="body" size="sm" noMargin>
+        Time:
+      </Typography>
+      <input
+        style={{width: "120px", marginLeft: "10px"}}
+        type="time"
```

Comment:
Question; have we tested the experience of the `time` input on various platforms and devices?